src/AWS.Deploy.Common/Exceptions.cs

```diff
@@ -126,6 +127,19 @@ public class OptionSettingItemDoesNotExistException : Exception
         public OptionSettingItemDoesNotExistException(string message, Exception? innerException = null) : base(message, innerException) { }
     }
 
+    /// <summary>
+    /// Thrown if <see cref="OptionSettingItem.SetValueOverride"/> is given an invalid value.
+    /// </summary>
+    public class ValidationFailedException : Exception
+    {
+        public ValidationResult ValidationResult { get; set; }
+
+        public ValidationFailedException(ValidationResult validationResult) : base(validationResult.ValidationFailedMessage)
```

Comment:
Made it consistent with rest of the exceptions. Note that this is not decorated as an expected exception because we are not handling it in the `CommandFactory` class